### PR TITLE
refactor: adapted identity module to satisfy conventions

### LINKED ISSUE #14: task: refactor and test identity module
### Description:

Refactor the identity module of the mainboard and add the missing unit tests.

### Implementation steps:

1. Change all the identifiers name following the embedded conventions
2. Implement unit tests
3. Fix small issues if any

### Dependencies:

None

### Additional notes:

_No response_

### PR DESCRIPTION
**Pull request by ale-giustina**
# Purpose
Refactoring of identity module

# Overview
Removed typedef on structs and enums, added doxygen comments and added unit tests.

# Guidance
The main files that have been affected are identity.h, identity-api.h, identity-api.c and test-identity.c. Other files may have been modified marginally due to variable renaming.

Closes #14

### CODE REVIEW

File: Core/Inc/bms/identity/identity.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#include "primary_network.h"
+#include "bms_network.h"
+
+/*!
+ * \brief Definition of the string containing the latest build time
+ *
+ * \details The build time string is in the following format %b %d %Y %H:%M:%S
+ * -> month day year hours:minutes:seconds (e.g. Apr 28 2024 11:45:15)
+ */
+#define IDENTITY_BUILD_TIME_STR (__DATE__ " "__TIME__)
```
Comment (Tonidotpy) [inline Core/Inc/bms/identity/identity.h:27]:
```suggestion
#define IDENTITY_BUILD_TIME_STR (__DATE__" "__TIME__)
```

File: Core/Inc/bms/identity/identity-api.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+/*!
+ * \brief Handle the received cellboard version
+ *
+ * \param payload A pointer to the canlib payload
+ */
+void identity_api_cellboard_version_handle(bms_cellboard_version_converted_t *payload);
+
+#else // CONF_IDENTITY_MODULE_ENABLE
+
+#define identity_api_init(id) CELLBOARD_NOPE()
```
Comment (Tonidotpy) [inline Core/Inc/bms/identity/identity-api.h]:
```suggestion
#define identity_api_init(id) EAGLETRT_API_NOP()
```

File: Core/Inc/bms/identity/identity-api.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#define identity_api_get_build_time() (0U)
+#define identity_api_get_mainboard_version_payload(byte_size) (NULL)
+#define identity_api_get_cellboard_version_payload(byte_size) (NULL)
+#define identity_api_cellboard_version_handle(payload) MAINBOARD_NOPE()
```
Comment (Tonidotpy) [inline Core/Inc/bms/identity/identity-api.h]:
```suggestion
#define identity_api_cellboard_version_handle(payload) EAGLETRT_API_NOP()
```

File: Core/Src/bms/identity-api.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+#include "eagletrt-api.h"
+
+#ifdef CONF_IDENTITY_MODULE_ENABLE
+
+EAGLETRT_STATIC struct IdentityHandler identity_handler;
+
+void identity_api_init(void) {
+    memset(&identity_handler, 0U, sizeof(identity_handler));
+
+    // TODO: Build time at compile time
```
Comment (Tonidotpy) [inline Core/Src/bms/identity-api.c:23]: I put this comment but it's probably not a good idea (maybe we could use a little bit of `constexpr` magic).
Right now it's not the priority so it can be left as it is.

File: test/test_identity/test-identity.c [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+#include "identity-api.h"
+
+#include <stdint.h>
+#include <string.h>
+
+extern struct IdentityHandler identity_handler;
+
+void test_identity_api_init(void) {
+    struct IdentityHandler expected_handler = { 0 };
+    // I am using this as it is just a getter
```
Comment (Tonidotpy) [inline test/test_identity/test-identity.c]: If you want to comment about a choice you made in the code put it on GitHub rather than inside the code, please.
```suggestion
```

File: test/test_identity/test-identity.c [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    for (CellboardId id = CELLBOARD_ID_0; id < CELLBOARD_ID_COUNT; ++id) {
+        expected_handler.cellboard_version_payload[id].cellboard_id = (primary_hv_cellboard_version_cellboard_id)id;
+    }
+
+    identity_api_init();
+
+    TEST_ASSERT_EQUAL_MEMORY_MESSAGE(&expected_handler, &identity_handler, sizeof(expected_handler), "Identity handler not initialized correctly");
+}
+
+void test_identity_api_get_mainboard_version_payload(void) {
```
Comment (Tonidotpy) [inline test/test_identity/test-identity.c:26]: Missing test for `get_mainboard_version_payload` with `NULL` byte size.

File: test/test_identity/test-identity.c [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    size_t byte_size = 0U;
+    primary_hv_mainboard_version_converted_t *payload = identity_api_get_mainboard_version_payload(&byte_size);
+
+    TEST_ASSERT_NOT_NULL_MESSAGE(payload, "Mainboard version payload pointer is NULL");
+    TEST_ASSERT_EQUAL_UINT32_MESSAGE(sizeof(identity_handler.mainboard_version_payload), byte_size, "Mainboard version payload byte size is incorrect");
+    TEST_ASSERT_EQUAL_MEMORY_MESSAGE(&identity_handler.mainboard_version_payload, payload, sizeof(identity_handler.mainboard_version_payload), "Mainboard version payload content is incorrect");
+}
+
+void test_identity_api_get_cellboard_version_payload(void) {
```
Comment (Tonidotpy) [inline test/test_identity/test-identity.c:42]: Missing test for `get_cellboard_version_payload` with `NULL` byte size.

**Review by Tonidotpy — CHANGES_REQUESTED** [review]

File: Core/Inc/bms/identity/identity.h (hunk 1 above)
Comment (ale-giustina) [reply Core/Inc/bms/identity/identity.h:27]: I didn't do it. I am innocent. It is clang tidy.

File: Core/Src/bms/identity-api.c (hunk 4 above)
Comment (ale-giustina) [reply Core/Src/bms/identity-api.c:23]: Issue created.

**Review by Tonidotpy — APPROVED** [review]
There are a couple of small errors that I'll fix during the merge.
Try to be tidier and more structured when you write code, don't jump from one function to another and mix the order of tests. Be consistent!!!
